eng/targets/Npm.Common.targets

```diff
@@ -53,6 +53,23 @@
 
     <Copy SourceFiles="$(PackageJson)" DestinationFiles="$(_BackupPackageJson)" />
 
+    <ReadLinesFromFile File="$(PackageJson)" Condition="@(ReplacePackageContent->Count()) != 0">
```

Comment:
cc @natemcmaster

Reply:
This is awesome @pranavkm. I'm not a huge fan of how we have to do it, but I'm glad to knock #10307 out as part of this.

For context @natemcmaster : Neither NPM nor Yarn seem to have any mechanism to have a "file" reference (i.e. Project Reference) and then convert it to a version reference during pack. So we have to do it in MSBuild...

There are tools ([Lerna](https://github.com/lerna/lerna)) that can make this better, but they are complicated and I don't think adding new tools is a great idea ;).

Reply:
Unfortunately, I don't think this will work on macOS/Linux due to https://github.com/microsoft/msbuild/issues/1024.

Minimal repro of the problem -- run this on Linux. The output has the backslash flipped to a forward slash. `{ "key": "value /" with a quote" }`.
```xml
<Project>
    <Target Name="Build">
        <PropertyGroup>
            <Content>{ "key": "value \" with a quote" }</Content>
        </PropertyGroup>

        <WriteLinesToFile Lines="$(Content)" File="out1.txt" Overwrite="true" />
    </Target>
</Project>
```

Reply:
Is there a way to write custom tasks with arcade? Additionally, could this wait for the next release? I don't think we publish npm packages built from xplat environments to npmjs, do we?

Reply:
Arcade doesn't support a 'repotasks' convention. We don't publish npm packages from xplat, but we would like for the repo to build correctly on all platforms.